docs/adding_software/adding_development_software.md

```diff
@@ -0,0 +1,155 @@
+
+``` python
+easyblock = 'CMakeMake'
+
+name = 'ESPResSo'
+version = 'devel'
+versionsuffix = '-%(software_commit)s'
+
+homepage = 'https://espressomd.org/wordpress'
+description = """A software package for performing and analyzing scientific Molecular Dynamics simulations."""
+
+github_account = 'espressomd'
+source_urls = ['https://github.com/%(github_account)s/%(name)s/archive/']
+
+sources = ['%(software_commit)s.tar.gz']
+```
+
+!!! warning "`--software-commit` disables `--robot`"
+
+    Using `--software-commit` disables the use of `--robot`, so make sure that you explicitly include
+    new dependencies that might need to be installed. Otherwise, the easyconfig files won't be found.
+
+One can also make new changes to the easyconfig file, for example, if the new functionality requires new build or 
+runtime dependencies, patches, configuration options, etc. It's a good idea to try installing from a specific commit locally first,
+to at least see if everything is parsed correctly and confirm that the right sources are being downloaded.
+
+
+While the process to build for `dev.eessi.io` is similar to the one for the [production repository](../repositories/software.eessi.io.md) there 
+are a few additional details to keep in mind.
+
+#### Versions
+
+Installations to the EESSI production repository refer to specific versions of applications, however, Development builds can't follow the same 
+approach as they are most often not pegged to a release. Because of this, it is possible to add a descriptive "version" name to the `version` field
+in the easyconfig file for a given (set of) installations. 
+
+Note that some applications are built with custom easyblocks which may
+use the `version` field to determine how the installation is meant to work (e.g., versions >2 need to copy files from to a new directory). Make sure
+that you account for this, otherwise you may install software differently than intended. If you encounter issues, you can open an issue in our
+[support portal](https://gitlab.com/eessi/support#eessi-support-portal).
+
+#### Installing dependencies
+
+Installations in `dev.eessi.io` are done _on top_ of `software.eessi.io`. That means if your development build depends on some application that is
+already installed in `software.eessi.io`, then that will simply be used. However, if you need to add a new dependency, then this must included as 
+part of the build. That means including an easystack file for it, and adding it to the right easystack file.
+
+#### Using commit IDs or tags for `--software-commit`
+
+Installing with `--software-commit` requires that you include either a commit ID or a tag. The installation procedure will use the commit ID or tag to 
+obtain the source from the right state. Because tags can be changed to point to a different commit ID, we recommend you avoid using them but stick with
```

Comment:
```suggestion
obtain the sources for the build. Because tags can be changed to point to a different commit ID, we recommend you avoid using them and sticking to